### PR TITLE
Only create local tracks if component is active

### PR DESCRIPTION
**Pull request by djee-ms**
Avoid creating a local audio or video track (webcam/microphone) if the
component is not active. Instead leave the Track property as NULL.

### CODE REVIEW

File: libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/Media/AudioSender.cs [hunk 1]
```diff
@@ -197,8 +197,9 @@ internal override async Task AttachTrackAsync()
             //< FIXME - Multi-track override!!!
             Transceiver.PeerConnection.PreferredAudioCodec = PreferredAudioCodec;
 
-            // Ensure the local sender track exists
-            if (Track == null)
+            // Ensure the local sender track exists, but do not create
+            // if the component is not active.
+            if ((Track == null) && isActiveAndEnabled)
```
Comment (fibann) [inline libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/Media/AudioSender.cs:202]: nit: if the component is enabled after `AttachTrackAsync` I suppose we should create the track in `OnEnable`. Or at least we should ensure that what happens if one does that is deterministic and documented.

**Comment by djee-ms** [comment]
Rebased, handled more corner cases, and added a bunch of Unity tests to validate all of that. Can you please re-review?

File: libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity.Tests.Runtime/VideoSenderTests.cs [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            Assert.IsNull(sender.Track);
+            Assert.IsFalse(sender.IsCapturing);
+
+            // Manually start capture
+            await sender.StartCaptureAsync();
+            Assert.IsNotNull(sender.Track);
+            Assert.IsTrue(sender.IsCapturing);
+        }
+
+        [Test(Description = "Capture can be manyally started even if not active.")]
```
Comment (fibann) [inline libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity.Tests.Runtime/VideoSenderTests.cs:121]:
```suggestion
        [Test(Description = "Capture can be manually started even if not active.")]
```

Comment (djee-ms) [reply libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity.Tests.Runtime/VideoSenderTests.cs]: Fixed.
